aircmd/core/artifact.py

```diff
@@ -30,48 +30,59 @@ async def build_task(pipeline: Pipeline, previous_result: PipelineResult) -> Pip
             #.with_exec(["poetry", "run", "ruff", "."])
             .with_exec(["poetry", "build"])
     )
-    await result.stdout()
-    pipeline_result = PipelineResult(status="success", data=result)
-    return pipeline_result
+    return result.sync()
 
-async def test_task(pipeline: Pipeline, build_result: PipelineResult) -> PipelineResult:
-    client = pipeline.client
+@task
+async def test_task(client: Client, build_result: Container) -> Container:
     mypy_cache: CacheVolume = client.cache_volume("mypy_cache")
     result = (with_poetry(client)
             .with_directory("/src", client.host().directory(".", include=["./tests", "./pyproject.toml", "./poetry.lock"]))
             .with_workdir("/src")
             .with_mounted_cache("/src/.mypy_cache", mypy_cache)
-            .with_directory("/src/dist", build_result.data.directory("/src/dist"))  # Mount the wheel directory
+            .with_directory("/src/dist", build_result.directory("/src/dist"))  # Mount the wheel directory
             .with_exec(["poetry", "install", "--only", "test"])  # Install the dependencies using Poetry
             .with_exec(["sh", "-c", "poetry run pip install $(find /src/dist -name 'aircmd-*.whl')"])  # Install the wheel file using Poetry
             .with_exec(["poetry", "run", "pytest"])  # Run the tests using Poetry
     )
-    await result.stdout()
-    pipeline_result = PipelineResult(status="success", data=result)
-    return pipeline_result
+    return result.sync()
 
 
+class BuildCommand(ClickCommandMetadata):
```

Comment:
@helderco I'm now coming more around to not separating this into its own class, from a usability perspective, but I think I'm going to revisit that once the activity around zenith dies down in case the API for entrypoints changes